plugin/src/ios/appDelegate.ts

```diff
@@ -0,0 +1,105 @@
+        /#import "AppDelegate.h"/g,
+        `#import "AppDelegate.h"
+#import <PushKit/PushKit.h>
+#import "RNVoipPushNotificationManager.h"
+#import "RNCallKeep.h"`
+      );
+    }
+
+    // Merging the method invocation block into the AppDelegate.m file
+    // having problem with auth,  https://github.com/react-native-webrtc/react-native-callkeep/issues/735
+    //try {
+    //  modResults.contents = mergeContents({
+    //    tag: "RNVoipPushNotificationAppDelegate",
+    //    src: modResults.contents,
+    //    anchor: methodInvocationLineMatcher,
+    //    offset: 0,
+    //    comment: "// ",
+    //    newSrc: methodInvocationBlock,
+    //  }).contents;
+    //} catch (e) {
+    //  // Fallback to the other regex
+    //  modResults.contents = mergeContents({
+    //    tag: "RNVoipPushNotificationAppDelegate",
+    //    src: modResults.contents,
+    //    anchor: fallbackInvocationLineMatcher,
+    //    offset: 0,
+    //    comment: "// ",
+    //    newSrc: methodInvocationBlock,
+    //  }).contents;
+    //}
+
+    // Add PushKit delegate method to the bottom of the file
+    // if other appDelegates are being implemented I will need to add this to the bottom of the file
+    if (!modResults.contents.includes("/* Add PushKit delegate method */")) {
+      modResults.contents = modResults.contents.replace(
+        /@end/g,
+        `/* Add PushKit delegate method */
+- (void)pushRegistry:(PKPushRegistry *)registry didUpdatePushCredentials:(PKPushCredentials *)credentials forType:(PKPushType)type
+{
+    [RNVoipPushNotificationManager didUpdatePushCredentials:credentials forType:(NSString *)type];
+}
+
+- (void)pushRegistry:(PKPushRegistry *)registry didInvalidatePushTokenForType:(PKPushType)type
+{
+
+}
+
+- (void)pushRegistry:(PKPushRegistry *)registry didReceiveIncomingPushWithPayload:(PKPushPayload *)payload forType:(PKPushType)type withCompletionHandler:(void (^)(void))completion
+{
+    NSString *uuid = payload.dictionaryPayload[@"uuid"];
+    NSString *callerName = [NSString stringWithFormat:@"%@ is Calling", payload.dictionaryPayload[@"callerName"]];
+    NSString *handle = payload.dictionaryPayload[@"handle"];
+    BOOL isVideo = [payload.dictionaryPayload[@"isVideo"] boolValue];
+    BOOL videoVal = NO;
+
+    if(isVideo) {
+      videoVal = YES;
+    }
+
+    [RNVoipPushNotificationManager addCompletionHandler:uuid completionHandler:completion];
+
+    [RNVoipPushNotificationManager didReceiveIncomingPushWithPayload:payload forType:(NSString *)type];
+
+    [RNCallKeep reportNewIncomingCall: uuid
+                               handle: handle
+                           handleType: @"generic"
+                             hasVideo: videoVal
+                  localizedCallerName: callerName
+                      supportsHolding: YES
+                         supportsDTMF: YES
+                     supportsGrouping: YES
+                   supportsUngrouping: YES
```

Comment:
Looks like these fields could be populated by payload as well. It would allow notification sender to set these options.